src/DurableTask.AzureStorage/Storage/DurableTaskStorageException.cs

```diff
@@ -35,12 +35,12 @@ public DurableTaskStorageException(string message, Exception inner)
         }
 
         public DurableTaskStorageException(RequestFailedException requestFailedException)
```

Comment:
Should we make this nullable to clarify our expectations about this parameter?
```suggestion
        public DurableTaskStorageException(RequestFailedException? requestFailedException)
```

Reply:
I think that makes sense. I updated the other arguments to match `Exception` in .NET 8: https://learn.microsoft.com/en-us/dotnet/api/system.exception.-ctor?view=net-8.0